sycl/test-e2e/Matrix/SPVCooperativeMatrix/SG32/get_coord_int8_matB.cpp

```diff
@@ -0,0 +1,21 @@
+//==----------- get_coord_int8_matB.cpp  - DPC++ joint_matrix---------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+// SG size = 32 is not currently supported for SYCL Joint Matrix by IGC on DG2
+// UNSUPPORTED: gpu-intel-dg2
+// REQUIRES: gpu, aspect-ext_intel_matrix
+// REQUIRES-INTEL-DRIVER: lin: 27501, win: 101.4943
+
+// RUN: %{build} -D__SPIRV_USE_COOPERATIVE_MATRIX -o %t.out
+// RUN: %{run} %t.out
+// XFAIL: cpu
```

Comment:
If there is a plan to enable these tests on CPU later, maybe it is better to use XFAIL: cpu everywhere instead of REQUIRES: gpu. That way, if we forget to enable some tests later, we will see XPASS. Otherwise, they will be skipped silently.

Reply:
`XFAIL: cpu` means - run it on CPU, expect run to finish and result to mismatch with the expected result. In case if CPU compiler crash - the test would be reported to be failed even with XFAIL. To avoid this one should instead use: RUN: not %command%, but IMHO it's effectively the same as to say REQUIRES: gpu

Reply:
hmm, from what I saw, when compiler can not compile the code (due to `internal error` or other compilation failures), it exits with some error code, and test is reported as xfail. Would you be able to try this with one test? If it is indeed reports failure instead of XFAIL, I'm fine with using `REQUIRES: gpu`.

Reply:
That was my understanding, for example that's how would work for LLVM's opt, clang or translator. May be it's different for E2E tests as in case of internal compiler error/crash it would be caught by SYCL runtime and turn into a proper exit code for LIT. Right now SPR machine I'm usually work with is down and setting up the emulator would take some time, so let me try to debug this using intel/llvm CI. Note, I currently set all tests to XFAIL on CPU, while I do expect many of them to actually pass. 

Also there is some inconsistency across the existing tests, some of them are having UNSUPPORTED: cpu, some XFAIL: cpu, some keeps REQUIRES: gpu. Probably we should align on that or is there some reasoning for it? For example I would expect out-of-bounds feature be supported on both CPU and GPU, but right now it's UNSUPPORTED (I can understand why it would fail - due to lacking CPU drivers in intel/llvm CI, but following your logic it should be then XFAILed).

Reply:
Seems like no cpu test runs in intel/llvm CI

Reply:
> May be it's different for E2E tests as in case of internal compiler error/crash it would be caught by SYCL runtime and turn into a proper exit code for LIT.

Yes, I think this is what happens.

> Also there is some inconsistency across the existing tests, some of them are having UNSUPPORTED: cpu, some XFAIL: cpu, some keeps REQUIRES: gpu. Probably we should align on that or is there some reasoning for it?

I expect that for most of them there are reasons. I admit that some of them maybe not aligned.  We can review case by case. If something is not aligned, I agree, we should align.

> I would expect out-of-bounds feature be supported on both CPU and GPU, but right now it's UNSUPPORTED (I can understand why it would fail - due to lacking CPU drivers in intel/llvm CI, but following your logic it should be then XFAILed).

I agree, XFAIL for cpu makes more sense to me. The feature maybe not yet implemented on CPU, but if we expect it to be implemented in the future, my preference would be XFAIL, so we catch if it starts to pass. 

> I'll also XFAILed tests that require W/A in the translator, will re-enable them along W/A PR landing

sounds good